addons/pos_self_order/controllers/orders.py

```diff
@@ -178,6 +178,18 @@ def get_tables(self, access_token):
 
         return tables
 
+    @http.route('/pos-self-order/make-payment-zero-amount-order', auth='public', type='json', website=True)
+    def make_payment_zero_amount_order(self, order_id):
+        # If order total amount is 0 then it will make a payment and convert
+        # order into paid without redirecting to payment screen.
+        order = request.env['pos.order'].sudo().browse(order_id)
```

Comment:
```suggestion
        # If order total amount is 0 then it will make a payment and convert
        # order into paid without redirecting to payment screen.
        pos_config = self._verify_pos_config(access_token)
        order = pos_config.env['pos.order'].browse(order_id)
```

Reply:
nice